src/main/java/gregtech/loaders/recipe/FuelRecipes.java

```diff
@@ -64,31 +63,69 @@ public static void registerFuels() {
         registerSemiFluidGeneratorFuel(FishOil.getFluid(8), 1, LV);
 
         //plasma turbine
-        registerPlasmaFuel(Helium.getPlasma(1), 2560, LV);
-        registerPlasmaFuel(Oxygen.getPlasma(1), 3072, LV);
-        registerPlasmaFuel(Nitrogen.getPlasma(1), 4096, LV);
-        registerPlasmaFuel(Iron.getPlasma(1), 6144, LV);
-        registerPlasmaFuel(Nickel.getPlasma(1), 12288, LV);
+        registerPlasmaFuel(Helium.getPlasma(1), Helium.getFluid(1), 2560, LV);
+        registerPlasmaFuel(Oxygen.getPlasma(1),  Oxygen.getFluid(1),3072, LV);
+        registerPlasmaFuel(Nitrogen.getPlasma(1), Nitrogen.getFluid(1), 4096, LV);
+        registerPlasmaFuel(Iron.getPlasma(1), Iron.getFluid(1), 6144, LV);
+        registerPlasmaFuel(Nickel.getPlasma(1), Nickel.getFluid(1), 12288, LV);
 
     }
 
     public static void registerPlasmaFuel(FluidStack fuelStack, int duration, int tier) {
-        RecipeMaps.PLASMA_GENERATOR_FUELS.addRecipe(new FuelRecipe(fuelStack, duration, GTValues.V[tier]));
+        RecipeMaps.PLASMA_GENERATOR_FUELS.recipeBuilder()
+                .fluidInputs(fuelStack)
+                .duration(duration)
+                .EUt((int) GTValues.V[tier])
+                .buildAndRegister();
+    }
+
+    public static void registerPlasmaFuel(FluidStack fuelStack, FluidStack fluidOutputStack, int duration, int tier) {
```

Comment:
This should validate if the passed fuelStack is actually a plasma

Reply:
Generators should not force fluids to be materials. I have no idea how checking whether fluids are materials works if another mod registers a fluid with the same name, and that fluid is used instead by forge. Furthermore, the original turbines accepted any fluid, so I don't see why these should not either. Mods like NuclearCraft also add generic "plasma" fluid, which would be potentially useful to use in this turbine as well.